### ISSUE TITLE
display sigma by default in `model_parameters()`

### ISSUE THREAD

**Issue by strengejacke** [easystats/parameters#305]
> I'm not sure it's necessary to display it by default in `model_parameters()` (I think that most people would _not_ expect it in the default parameters)

Why not? This is nothing particular new and always worked for Bayesian mixed models, now it works for non-mixed as well. I'm currently reading https://avehtari.github.io/ROS-Examples/, and this convinced me including information on residual standard deviation as default for `model_parameters()`:

``` r
library(parameters)
model <- lm(mpg ~ wt + cyl, data = mtcars)

model_parameters(model)
#> Parameter   | Coefficient |   SE |         95% CI |     t | df |      p
#> -----------------------------------------------------------------------
#> (Intercept) |       39.69 | 1.71 | [36.18, 43.19] | 23.14 | 29 | < .001
#> wt          |       -3.19 | 0.76 | [-4.74, -1.64] | -4.22 | 29 | < .001
#> cyl         |       -1.51 | 0.41 | [-2.36, -0.66] | -3.64 | 29 | 0.001 
#> 
#> Residual standard deviation: 2.57
```

<sup>Created on 2020-09-23 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

But I'm open for other suggestions.

_Originally posted by @strengejacke in https://github.com/easystats/bayestestR/issues/338#issuecomment-697249869_

**Comment by mattansb** [comment]
I say no - it's currently breaking some `effectsize` code :/

**Comment by strengejacke** [comment]
How's that? It only affects the `print()` method.

**Comment by mattansb** [comment]
`model_parameter` for Bayesian models has started returning `sigma` parameter... Is this not the source of that?

**Comment by strengejacke** [comment]
Ah, ok. For Bayesian models. But that was intentional? See https://github.com/easystats/bayestestR/issues/338

**Comment by strengejacke** [comment]
> I say no - it's currently breaking some `effectsize` code :/

Can you point out to where it breaks code? We may add an if-statement, checking the _insight_ version, so it works both with current and future _insight_, until we can remove this in one of the next update rounds.

**Comment by DominiqueMakowski** [comment]
> and this convinced me including information on residual standard deviation as default for model_parameters()

@strengejacke I'd be very to hear in a condensed summarized summary about why is it important (TBH I'm clueless as to its interpretation, so don't know what to "care about " when I look at this value) ^^

**Comment by IndrajeetPatil** [comment]
> > and this convinced me including information on residual standard deviation as default for model_parameters()
> 
> @strengejacke I'd be very curious to hear in a condensed summarized summary about why is it important (TBH I'm clueless as to its interpretation, so don't know what to "care about " when I look at this value) ^^

Ditto!

I am also wondering if this is something that should be included in the `model_performance` output?

**Comment by strengejacke** [comment]
- `get_parameters()` for Sta-models has always returned sigma for mixed models.
- For non-mixed Stan-models, `get_parameters()` did not include sigma, however, this was requested here: https://github.com/easystats/bayestestR/issues/338 - so it's now consistent with "Stan mixed models". That's the reason for the change in _insight_
- Other functions / code is not affected, since non-Bayesian models only add sigma as attribute in `model_parameters()`, and this is only used for `print()`.

The rationale for having an interest in the sigma parameter is summarized here:
https://easystats.github.io/insight/reference/get_sigma.html

**Comment by DominiqueMakowski** [comment]
> I am also wondering if this is something that should be included in the model_performance output?

It's true that it seems to be more related to model properties than to its parameters per se

**Comment by strengejacke** [comment]
ok, I will keep the `show_sigma` argument, but it defaults to `FALSE`, and `model_performance()` may get an additional "sigma" column?

**Comment by mattansb** [comment]
Sounds good.

The code breaks are not on CRAN (as all Bayesian tests are skipped on CRAN 😅), but I'm suddenly getting an extra row.
Can the `show_sigma` (`show_aux`?) be set to `FALSE` in `bayestestR` as well?

**Comment by strengejacke** [comment]
There is no `show_sigma` in _bayestestR_. Most bayestestR functions just wrap or rely on `get_parameters()` and process all parameters, no matter which one. We can distinguish between "fixed" and "random" (with the `effects` argument), or "all", "conditional" or "zi" (with `component`). Mostly, we use "all" by default.

So.. you opened the Pandora's box by asking:
> is there a way to get _all_ of the model's parameters?

Now you must deal with it. 😄

**Comment by strengejacke** [comment]
Sigma is one of the most common "auxiliary" parameters, so unless we have a more clear idea of how to access auxiliary parameters in general, and whether to return them in `get_parameters()` or not (see also https://github.com/easystats/insight/issues/230), I'd vote for including it by default in `get_parameters()` for Bayesian non-mixed, as we already did for Bayesian mixed.

**Comment by mattansb** [comment]
Damn me and my questions....

**Comment by DominiqueMakowski** [comment]
> ok, I will keep the show_sigma argument, but it defaults to FALSE, and model_performance() may get an additional "sigma" column?

Yes and yes, even though I'm still not convinced by having it by default (maybe make it optional for now so that we can safely go ahead with this round of updates and slowly see what will it affect to return sigma). 

Thinking of report, I still do think that it will be something good if we first manage to get users to correctly interpret/understand a handful of core indices... rather than to prompt them to report as many of indices as they can without understanding and thinking about them. And since `sigma` doesn't seem to be as central as compared to others in that regards... I'd say it's unnecessary to clutter the output by default with it 😁 (I mean I did read with attention your details about sigma, but it still didn't strike me as a critical index which absence in the description of a model would wreak complete havoc)

**Comment by mattansb** [comment]
Sigma is related to RMSE, so I like the idea of having sun aux parameters also in performance, maybe?

As for report - I agree that as a model-parameter sigma is not very informative, especially in social-sci (when the over all scale is usually considered 😴).

**Comment by strengejacke** [comment]
I'm confused... 

1. sigma is _not_ reported by default in the `print()` for non-Bayesian models (so nothing changed so far)
2. sigma has _always_ been reported for Bayesian mixed models, so nothing has changed so far
3. sigma is _now additionally_ included for Bayesian non-mixed models, this is the only thing that has changed.

Bringing 2. and 3. in line seems to more consistent than the previous behaviour, where `get_parameters()` returned sigma for some Bayesian models, but not for all. This change should _not_ break any code in bayestestR or parameters, it may break tests (we need to check by running local checks with latest _insight_). Only exception seems to be _effectsize_. 

From all comments in https://github.com/easystats/bayestestR/issues/338, I understood that it makes sense to have consistent behaviour in `get_parameters()´ for Bayesian models. To achieve this, we had two options for `get_parameters()`: remove sigma from mixed, or add sigma to non-mixed. I read the 338-issue that it's preferred to add sigma to non-mixed...?

**Comment by strengejacke** [comment]
> I agree that as a model-parameter sigma is not very informative

After reading Gelman, Hill and Vehtari, I'd disagree. There's a clear and straightforward interpretation in terms of the scale of the response.

**Comment by DominiqueMakowski** [comment]
> There's a clear and straightforward interpretation in terms of the scale of the response.

But is it any more useful / more intuitive / critically complimentary than the other existing indices like R2 RMSE or so?

**Comment by strengejacke** [comment]
> > There's a clear and straightforward interpretation in terms of the scale of the response.
> 
> But is it any more useful / more intuitive / critically complimentary than the other existing indices like R2 RMSE or so?

![sigma](https://user-images.githubusercontent.com/26301769/94373802-0af31100-0108-11eb-8bf2-9b06bb2c14f6.png)

Compared to RMSE: sigma is the SD, (R)MSE the average square (root) of the residuals. So, these indices are all related because based on residuals, but they are slightly different regarding their interpretation.

**Comment by DominiqueMakowski** [comment]
aah interesting thanks @strengejacke it's clearer now ☺️
